docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,489 @@
+            "PORTIN",
+            "PORTINOUT",
+            "PORTOUT",
+            "PWMOUT",
+            "RESET_REASON",
+            "SERIAL",
+            "SERIAL_FC",
+            "SERIAL_ASYNCH",
+            "SLEEP",
+            "SPI",
+            "SPI_ASYNCH",
+            "SPISLAVE",
+            "STDIO_MESSAGES",
+            "TRNG",
+            "FLASH",
+            "USBDEVICE",
+            "WATCHDOG"
+        ],
+        "release_versions": [
+            "2",
+            "5"
+        ],
+        "device_name": "MK64FN1M0xxx12",
+        "bootloader_supported": true,
+        "overrides": {
+            "network-default-interface-type": "ETHERNET"
+        },
+        "supported_c_libs": {
+            "arm": [
+                "std",
+                "small"
+            ],
+            "gcc_arm": [
+                "std",
+                "small"
+            ],
+            "iar": [
+                "std"
+            ]
+        }
+    }
+```
+
+Note the config options `macros_add`, `extra_labels_add` and `device_has_add`. To add or remove `macros`, `extra_labels` or target capabilities, a PSA target definition **must** use [macros/extra_labels/device_has]`_add` or [macros/extra_labels/device_has]`_remove` (not `macros`, `extra_labels` or `device_has`).
+
+To add or remove a feature, use `feature_`[add/remove].
+
+As a second example, here is a single-core Armv7-M PSA target that inherits from one of its family targets:
+
+```json
+    "GD32_F450ZI": {
+        "inherits": [
+            "GD32_Target"
+        ]
+    }
+```
+
+In this case, you must add the following additional attributes:
+
+```json
+        "features_add": [
+            "PSA"
+        ],
+        "extra_labels_add": [
+            "MBED_PSA_SRV"
+        ]
+```
+
+### Adding dual-core PSA targets
+
+A target can be categorized as a dual-core target if it has at least two cores that are either Armv7-M or Armv6-M. On dual-core PSA targets, TF-M runs on the SPE and provides PSA services.
+
+An Mbed OS (NSPE) target must contain the following attributes (in addition to other target attributes defined in [porting a custom board](../porting/porting-a-custom-board.html) and [porting targets](../porting/porting-targets.html)):
+
+* `inherits`: PSA generic target `PSA_DUAL_CORE`, unless the target has to inherit
+  from one of its family targets.
+* `tfm_target_name`: Target name in TF-M.
+* `tfm_bootloader_supported`: If TF-M bootloader is supported by the target.
+
+    The supported values are `true` and `false`.
+* `tfm_supported_toolchains`: Supported TF-M toolchains.
+
+    The supported values are `ARMCLANG` and `GNUARM`.
+* `tfm_default_toolchain`: Default TF-M toolchain.
+
+    The supported values are `ARMCLANG` and `GNUARM`.
+* `tfm_delivery_dir`: The directory to which TF-M binaries will be copied.
+* `TFM_OUTPUT_EXT`: Optional attribute that indicates the output extension of the TF-M secure binary.<!--how come only this one is all caps?-->
```

Comment:
Do we want to remove the `<!--how come only this one is all caps?-->` or other `<!--` text from the markdown for resolved issues?

Reply:
Yes, though if we miss one it's not the end of the world as it doesn't get rendered.